mobile/docs/release.md

```diff
@@ -0,0 +1,32 @@
+# Releases
+
+Create a PR to bump up the version in `pubspec.yaml`.
+
+> [!NOTE]
+>
+> Use [semver](https://semver.org/) for the tags, with `photos-` as a prefix.
+> Multiple beta releases for the same upcoming version can be done by adding
+> build metadata at the end, e.g. `photos-v1.2.3-beta+3`.
+
+Once that is merged, tag main, and push the tag.
+
+```sh
+git tag photos-v1.2.3
+git push origin photos-v1.2.3
+```
+
+This'll trigger a GitHub workflow that:
+
+* Creates a new draft GitHub release and attaches the build artifacts to it
```

Comment:
@vishnukvmd I've kept this document separate instead of linking to the auth's docs, even though it is mostly a copy paste. This is because the release processes are divergent currently. We can also document the f-droid stuff here (one liners are enough, just how the f-droid build gets triggered) and mention how people can verify the authenticity of the build by looking for the sha in the GitHub workflow logs. Then we can just give out a link to this doc for people looking for build authenticity.